var/spack/repos/builtin/packages/py-numpy/package.py

```diff
@@ -166,6 +157,27 @@ def setup_environment(self, spack_env, run_env):
 
         run_env.prepend_path('CPATH', include_path)
 
+    # Do the usual with gcc
+    def get_phases(self):
```

Comment:
I wrote this function because I wanted to build and install in one step only for intel compilers. If using GCC/Clang, do the usual and if using Intel, build and install in one step. Is there a better way to do this ?

Reply:
I don't know of any package that does this. If it's possible, it would look something like this:
```python
@property
def phases(self):
    if self.spec.satisfies('%intel'):
        return [...]
    else:
        return [...]
```
@alalazo might know if this is kosher or not. Do you _have_ to use different phases? Could you use the same phases for both?

Reply:
I'm just doing what the [docs](https://docs.scipy.org/doc/scipy/reference/building/linux.html#intel-c-compiler-and-mkl) ask me to with no modifications whatsover so yes. If that has to be avoided then someone needs to check it by breaking it into multiple steps and checking the libraries being linked at each stage. That would have the same multimethods issue where the build and install args would be different depending on the compiler used.

Reply:
Does having dynamic phases break anything ? If it doesn't then this approach is the best because it's better not to mess with whatever assumptions the people who develop `numpy.disutils` might have taken rather than force static phases only to discover that some assumption was violated (which might not be the case but I wonder if it's worth the effort).

Reply:
Does having dynamic phases break anything ? If it doesn't then this approach is preferable because it's better not to mess with whatever assumptions the people who develop `numpy.disutils` might have taken rather than force static phases only to discover that some assumption was violated (which might not be the case but I wonder if it's worth the effort).